lib/server.go

```diff
@@ -298,6 +315,16 @@ func (srv *Server) StartIptables() error {
 		return fmt.Errorf("failed to add SNAT rule: %v", err)
 	}
 
+	err = srv.iptablesMssRule(true)
+	if err != nil {
+		if err := srv.iptablesSnatRule(false); err != nil {
+			return fmt.Errorf("failed to add MSS rule: %v", err)
+		}
+		if err := srv.iptablesInputFwmarkRule(false); err != nil {
+			return fmt.Errorf("failed to add MSS rule: %v", err)
```

Comment:
I think it's better to disable both rules (as many as possible) and then return fmt.Errorf at the end regardless of whether the disable succeeded or failed.